### ISSUE TITLE
Leading semicolons in if-statements break highlighting

### ISSUE THREAD

**Issue by lourd** [babel/babel-sublime#319]
Using v8.6.3, Sublime build 3126, OSX 10.12.3

I stumbled across this case of broken highlighting when adding Prettier to my project with semicolon-less style. When using do expressions within JSX, Prettier adds a leading semicolon to the source, i.e.

```js
do {
  if (true) {
    ;<span>Foo</span>
  } else {
    ;<span>Bar</span>
  }
}
```

This causes the highlighting of all the code beyond the semicolon to be messed up.

**Comment by borela** [comment]
Try FJSX15 https://github.com/borela/naomi

**Comment by Thom1729** [comment]
This should be fixed in v10.0.0-beta.1. See #379 for more information on the new release. Please report any problems with v10 in a new issue.
